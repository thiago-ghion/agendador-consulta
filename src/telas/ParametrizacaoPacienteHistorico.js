import React from "react";
import { Button } from "react-bootstrap";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

function ParametrizacaoPacienteHistorico(props) {
  return (
    <div>
      <br></br>
      <Row className="justify-content-md-end">
        <Col md={1}>
          <Button variant="primary" onClick={() => props.setTelaAtiva(1)}>
            Voltar
          </Button>
        </Col>
      </Row>
    </div>
  );
}

export default ParametrizacaoPacienteHistorico;
