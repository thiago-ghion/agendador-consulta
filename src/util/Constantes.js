const URL_API = 'http://localhost:5000';

export { URL_API };