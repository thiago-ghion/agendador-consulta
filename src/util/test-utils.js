import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
// As a basic setup, import your same slice reducers
import loginReducer from "../features/loginSlice";
import profissionalReducer from "../features/profissionalSlice";
import consultaReducer from "../features/consultaSlice";
import pacienteReducer from "../features/pacienteSlice";
import horarioReducer from "../features/horarioSlice";
import usuarioReducer from "../features/usuarioSlice";
import loadingReducer from "../features/loadingSlice";
import mensagemgReducer from "../features/mensagemSlice";

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: {
        login: loginReducer,
        consulta: consultaReducer,
        profissional: profissionalReducer,
        paciente: pacienteReducer,
        horario: horarioReducer,
        usuario: usuarioReducer,
        loading: loadingReducer,
        mensagem: mensagemgReducer,
      },
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
