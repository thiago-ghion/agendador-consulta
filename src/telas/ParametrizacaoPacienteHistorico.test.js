import React from "react";
import { rest } from "msw";
import { setupServer } from "msw/node";
import { renderWithProviders } from "../util/test-utils";

import ParametrizacaoPacienteHistorico from "./ParametrizacaoPacienteHistorico";
import { fireEvent, screen, waitFor } from "@testing-library/react";

export const handlers = [];

const server = setupServer(...handlers);

// Enable API mocking before tests.
beforeAll(() => server.listen());

// Reset any runtime request handlers we may add during the tests.
afterEach(() => server.resetHandlers());

// Disable API mocking after the tests are done.
afterAll(() => server.close());

beforeEach(() => {});

test("Renderização componente", async () => {
  expect(
    renderWithProviders(<ParametrizacaoPacienteHistorico />)
  ).toMatchSnapshot();
});

test("Botão voltar", async () => {
  renderWithProviders(
    <ParametrizacaoPacienteHistorico setTelaAtiva={() => {}} />
  );
  const voltar = await waitFor(() =>
    screen.findByRole("button", { name: /Voltar/ })
  );
  fireEvent.click(voltar);
});
