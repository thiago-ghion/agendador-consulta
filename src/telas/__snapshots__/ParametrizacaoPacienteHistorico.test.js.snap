// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderização componente 1`] = `undefined`;
